core/common/src/main/java/alluxio/retry/RetryUtils.java

```diff
@@ -109,5 +137,61 @@ public static RetryPolicy defaultActiveSyncClientRetry(long activeUfsPollTimeout
     void run() throws IOException;
   }
 
+  /**
+   * Interface for methods which return a result and may throw IOException.
+   * @param <T> type of the return value
+   */
+  @FunctionalInterface
+  public interface RunnableFunctionThrowsException<T> {
+    /**
+     * Runs the runnable.
+     * @return T return an object of type T
+     */
+    T run() throws RetryException, CantRetryException;
+  }
+
+  /**
+   * Exceptions that can be retried.
+   */
+  public static class RetryException extends IOException {
```

Comment:
can we call this `RetryableException`, and make it its own class (in this directory)?